xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -1350,11 +1350,23 @@ bool CGUIWindowManager::Render()
   assert(CServiceBroker::GetAppMessenger()->IsProcessThread());
   CSingleExit lock(CServiceBroker::GetWinSystem()->GetGfxContext());
 
+  int bufferAge = CServiceBroker::GetWinSystem()->GetBufferAge() + 1;
+  bool visualizeDirtyRegions =
+      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_guiVisualizeDirtyRegions;
+  if (visualizeDirtyRegions)
+    bufferAge = 20;
+  if (bufferAge)
```

Comment:
The buffer age can be 0. To quote EGL_EXT_buffer_age:
```
    This extension lets applications query the age of the back buffer
    contents for an EGL surface as the number of frames elapsed since
    the contents were most recently defined. The back buffer can
    either be reported as invalid (has an age of 0) or it may be
    reported to contain the contents from n frames prior to the
    current frame.
```
This is the case after initializing the frame buffers (on application start, viewport change, etc).

Reply:
> This is the case after initializing the frame buffers (on application start, viewport change, etc).

OK, that means the `bufferAge` variable is always `>0` and the `if (bufferAge)` check is not needed.

Reply:
bufferAge can at any time become zero. One example is the mesa etnaviv driver. When it realizes that OpenGLES applications do not need all the performance it might reduce its internal buffer count from maybe 5 to 4, which would mean that more stages need to repaint. So whenever this happens, we have to get all three last blits (incase of triple buffer) on the screen.

Reply:
And the other way round. new buffer (in driver) means full repaint once.

Reply:
I've moved the `bufferAge + 1`, so the checks make sense. I've also limited the maximum tracked buffers, in case there is something up with the driver.

Reply:
Whoops, my second message got delayed. @fritsch, are you satisfied?

Reply:
Sure. Thx.

Reply:
I don't think we need the 20 ;-) but safe is safe. Imagine the amount am DDR that needs.

Reply:
The 20 is a bit much, but I have not touched the number. When debugging, I'm usually pinging between 1 and 5.